dashboard/modules/job/cli.py

```diff
@@ -123,6 +123,13 @@ def job_cli_group():
         "If specified, this overrides the option in `--runtime-env`."
     ),
 )
+@click.option(
+    "--metadata",
+    type=str,
+    default=None,
+    required=False,
+    help="JSON-serialized dictionary of metadata to attach to the job.",
```

Comment:
Q, wonder if it is possible to just pass --metadata="key=value" like the state API --filter option? Providing json serialize input is pretty cumbersome. cc @edoakes

Reply:
Agree, that sounds like a better option to me. I believe you can do `multiple=True` in the `click.option` and then support:
`ray job submit -m key1=val1 -m key2=val2 -- echo hello`

Reply:
That's fine w/ me, let's just rename the arg added here as `--metadata-json` then so we can use `--metadata` in the future :)